src/eosjs-api.ts

```diff
@@ -241,7 +247,14 @@ export class Api {
             if (!info) {
                 info = await this.rpc.get_info();
             }
-            const refBlock = await this.rpc.get_block_header_state(info.head_block_num - blocksBehind);
+
+            const taposBlockNumber = info.head_block_num - blocksBehind;
+            let refBlock: GetBlockHeaderStateResult | GetBlockResult;
+            if (taposBlockNumber - info.last_irreversible_block_num < 2) {
```

Comment:
There's a race condition. If this is false now, it may be true when nodeos processes the get_block_header_state  request. If that fails, it needs to retry with get_block.

Reply:
I'll just wrap in a try/catch instead of trying to predict when this error will occur then.